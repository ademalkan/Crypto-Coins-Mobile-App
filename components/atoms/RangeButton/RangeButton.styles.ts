import { StyleSheet } from "react-native";

export default StyleSheet.create({
  timeRangeButton: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    marginHorizontal: 8,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: '#ccc',
  },
  selectedTimeRangeButton: {
    backgroundColor: '#ccc',
  },
  buttonText: {
    fontSize: 14,
    fontWeight: 'bold',
    color: '#000',
  },
})