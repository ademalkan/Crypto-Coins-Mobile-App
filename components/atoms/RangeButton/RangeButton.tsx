import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import styles from './RangeButton.styles'
interface RangeButtonProps {
  label: string;
  onPress: () => void;
  selected?: boolean;
}

const RangeButton: React.FC<RangeButtonProps> = ({ label, onPress, selected }) => {
  return (
    <TouchableOpacity
      style={[styles.timeRangeButton, selected && styles.selectedTimeRangeButton]}
      onPress={onPress}
    >
      <Text style={styles.buttonText}>{label}</Text>
    </TouchableOpacity>
  );
};

export default RangeButton;
