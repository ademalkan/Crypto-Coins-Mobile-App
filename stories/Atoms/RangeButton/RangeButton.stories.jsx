import React from "react";
import { View } from "react-native";
import RangeButton from "../../../components/atoms/RangeButton/RangeButton";

export default {
  title: "RangeButton",
  component: RangeButton,
};

export const Default = () => (
  <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
    <RangeButton label="Button" onPress={() => {}} />
  </View>
);

export const Selected = () => (
  <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
    <RangeButton label="Button" onPress={() => {}} selected />
  </View>
);
