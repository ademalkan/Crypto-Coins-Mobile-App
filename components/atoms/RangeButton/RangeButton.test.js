import React from "react";
import renderer from "react-test-renderer";
import { TouchableOpacity } from "react-native";
import RangeButton from "./RangeButton";

describe("RangeButton", () => {
  it("should render the label prop correctly", () => {
    const label = "Button Label";
    const onPress = jest.fn();

    const component = renderer.create(
      <RangeButton label={label} onPress={onPress} selected={false} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call onPress when button is pressed", () => {
    const label = "Button Label";
    const onPress = jest.fn();

    const component = renderer.create(
      <RangeButton label={label} onPress={onPress} selected={false} />
    );

    const button = component.root.findByType(TouchableOpacity);
    button.props.onPress();

    expect(onPress).toHaveBeenCalled();
  });

  it("should apply selected style when selected prop is true", () => {
    const label = "Button Label";
    const onPress = jest.fn();

    const component = renderer.create(
      <RangeButton label={label} onPress={onPress} selected={true} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
