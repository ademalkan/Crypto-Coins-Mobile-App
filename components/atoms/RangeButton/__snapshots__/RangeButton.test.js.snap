// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeButton should apply selected style when selected prop is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ccc",
      "borderColor": "#ccc",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginHorizontal": 8,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Button Label
  </Text>
</View>
`;

exports[`RangeButton should render the label prop correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#ccc",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginHorizontal": 8,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Button Label
  </Text>
</View>
`;
